manta-js/package/sdk.ts

```diff
@@ -0,0 +1,626 @@
+            this.api.setSigner(polkadotSigner);
+
+        } catch (e) {
+            console.error("Unable to set polkadotJS signer.", e);
+        }
+    }
+
+    ///
+    /// SDK methods
+    ///
+
+    /// Convert a private address to JSON.
+    convertPrivateAddressToJson(address: string): any {
+        return privateAddressToJson(address);
+    }
+
+    /// Returns information about the currently supported networks.
+    networks(): any {
+        return config.NETWORKS;
+    }
+
+    /// Returns whether the current wasm wallet instance is busy. To avoid multiple
+    /// calls to wasmWallet simultaneously which causes a borrowMut rust error.
+    isWalletBusy(): boolean {
+        return this.walletIsBusy;
+    }
+
+    ///
+    /// Manta Signer methods
+    ///
+
+    /// Returns the zkAddress of the currently connected manta-signer instance.
+    async privateAddress(): Promise<Address> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            const privateAddress = await getPrivateAddress(this.wasm, this.wasmWallet, this.network);
+            this.walletIsBusy = false;
+            return privateAddress
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
+
+    /// Performs full wallet recovery. Restarts `self` with an empty state and
+    /// performs a synchronization against the signer and ledger to catch up to
+    /// the current checkpoint and balance state.
+    ///
+    /// Requirements: Must be called once after creating an instance of MantaPrivateWallet 
+    /// and must be called before walletSync(). Must also be called after every 
+    /// time the network is changed.
+    async initalWalletSync(): Promise<void> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            await initSync(this.wasm, this.wasmWallet, this.network);
+            this.walletIsBusy = false;
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
+
+    /// Pulls data from the ledger, synchronizing the currently connected wallet and
+    /// balance state. This method runs until all the ledger data has arrived at and
+    /// has been synchronized with the wallet.
+    async walletSync(): Promise<void> {
+        if (!this.walletIsBusy) {
+            this.walletIsBusy = true;
+            await sync(this.wasm, this.wasmWallet, this.network);
+            this.walletIsBusy = false;
+        } else {
+            console.error("Wallet is currently busy!");
+        }
+    }
+
+    /// Returns the version of the currently connected manta-signer instance.
+    async signerVersion(): Promise<Version> {
+        const version = await getSignerVersion();
+        return version;
+    }
+
+    ///
+    /// Fungible token methods
+    ///
+ 
+    /// Returns the metadata for an asset with a given `asset_id` for the currently
+    /// connected network.
+    async assetMetaData(assetId: BN): Promise<any> {
```

Comment:
Is there any reason assetId parameters throughout this file shouldn't just be a `number`? Would be more convenient, as we represent assetIds as `number`s on our front end.

Reply:
assetId is a BN because javascript "number" doesn't support u128 but on the runtime side assetId is u128.